src/View/ThemeResourceLoader.php

```diff
@@ -174,6 +181,12 @@ public function getPath($identifier)
      */
     public function findTemplate($template, $themes = null)
     {
+        // Look for a cached result for this data set
+        $cacheKey = json_encode($template) . json_encode($themes);
+        if (isset(static::$cacheData[$cacheKey])) {
+            return static::$cacheData[$cacheKey];
+        }
```

Comment:
Instead of storing these in memory per session can we implement a proper caching interface? At the very least, if we have an in-memory cache then we get the same results as this and at best we can use APCu and share the same cache for all requests.

Reply:
Note: You might want to ensure the cache isn't being applied in dev mode

Reply:
>Note: You might want to ensure the cache isn't being applied in dev mode

That'd probably be the same as the config/template/class manifests being cached, which devs are used to flushing when they change things. I don't think that's a problem here

Reply:
I've implemented the regular SS cache API instead of a static memory cache

Reply:
Part of me desperately wants to find an MD5 collision, here!!!